pentaho-platform-plugin-deployer/src/main/java/org/pentaho/osgi/platform/plugin/deployer/PlatformPluginBundlingURLConnection.java

```diff
@@ -105,15 +113,20 @@ public InputStream getInputStream() throws IOException {
       nameVersion = DeployerUtils.extractNameVersionType( mvnPath );
     }
 
+    //Check to see if the bundle is already installed
+    boolean isPluginProcessedBefore = bundleStateManager.isBundleInstalled( nameVersion[0] + nameVersion[1] );
```

Comment:
Can't we just check the BundleContext.getBundles() to see if it is already installed? Do we really need to maintain our own list?